src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionResolveHandler.cs

```diff
@@ -158,6 +178,18 @@ public async Task<CompletionItem> HandleRequestAsync(CompletionItem request, Cli
             }
 
             return resolvedCompletionItem;
+
+            static string ConvertEditIndentationToTabs(string textEdit, int tabSize)
+            {
+                var indentationString = new StringBuilder();
+                for (var i = 0; i < tabSize; i++)
+                {
+                    indentationString.Append(' ');
+                }
+
+                var updatedTextEdit = textEdit.Replace(indentationString.ToString(), "\t");
```

Comment:
It might not be a good idea to just use replace. For example this replacement on `    var myString = "I        like     lots    of     spaces";` would break user code. Maybe this isn't a problem with completion though?

Perhaps a better approach here would be to get the index of the first non-whitespace character then call [`FormattingContext.GetIndentationString`](https://github.com/dotnet/aspnetcore-tooling/blob/6c2182993d88fa925de9db2a1911ca18e78b68fb/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingContext.cs#L97) (perhaps moving it to a better spot), and then just replace those initial characters. If that makes sense?

Reply:
Thanks for the feedback! I tested with tabSize/indentSize set to 1 and everything still seems to work OK even with lots of spaces in the TextEdit. I guess the Razor formatter might be doing some magic behind the scenes to make everything magically work?

> Perhaps a better approach here would be to get the index of the first non-whitespace character then call FormattingContext.GetIndentationString (perhaps moving it to a better spot), and then just replace those initial characters. If that makes sense?

Ahh I was looking at this method, but I don't know if it would work here since (1) the whitespace before the first non-whitespace character isn't guaranteed to be indentation, and (2) there might be multiple indentations within the TextEdit. I definitely see where you're coming from, I don't think it's ideal to just use replace here, but I couldn't think of a better alternative.

Reply:
If it's crazy and it works, then it's not crazy. Also I just noticed GitHub swallowed lots of extra spaces in my example 🤦‍♂️

Good point about the whitespace not being indentation, I didn't think of that. Perhaps that, plus the fact that a completion item is unlikely to have strings with multiple spaces, mean the concern is moot.